src/backend/distributed/master/master_stage_protocol.c

```diff
@@ -601,11 +613,13 @@ WorkerTableSize(char *nodeName, uint32 nodePort, Oid relationId, char *tableName
 
 	if (cstoreTable)
 	{
-		appendStringInfo(tableSizeQuery, SHARD_CSTORE_TABLE_SIZE_QUERY, tableName);
+		appendStringInfo(tableSizeQuery, SHARD_CSTORE_TABLE_SIZE_QUERY,
+						 quote_literal_cstr(tableName));
```

Comment:
Maybe
quotedTableName = quote_literal_cstr(tableName);

and use quotedTableName in all 3 occurrences.